### ISSUE TITLE
[RFE] Ability to churn an existing workload

### ISSUE THREAD

**Issue by josecastillolema** [kube-burner/kube-burner#735]
**Is your feature request related to a problem? Please describe.**
When combining kube-burner with other perf tools, i.e.: data plane testing, I would like to use them simultaneously. However, I only want to start the data plane perf tool in the churn phase, not in the creation of the workload phase.

**Describe the solution you'd like**
I would like the ability to churn an existing workload, passing its ID.

**Comment by vishnuchalla** [comment]
@josecastillolema Can you please elaborate a bit on this idea? From the description it looks like something that can be achieved with a simple shell script. Why kube-burner?

Here is what I think.
for churn - a shell scirpt to toogle (on/off) the pid with certain wait times in between for certain number of iterations.
for metrics - use kube-burner index `sub-command` with start and end timestamps to collect metrics.

Can you please confirm or provide a detail example on this idea?

**Comment by vishnuchalla** [comment]
Closing the issue is it has been stale for a while and seems to be having an alternative to achieve the same behavior.

**Comment by sferlin** [comment]
I haven't seen this before, but I working this at the moment just for info.

I am orchestrating PIDs with a python script, e.g. the dataplane starts (it takes +3 min. for the dataplane workload to actually start, with crucible as an example, only after "client-start-end"), and then kube-burner is called. 

However, this initial overlap of both processes still includes 1) kube-burner startup phase, and 2) it does not consider the intervals (2min. is the default) between churn cycles. Cutting off the start-up phase and keeping only the overlap of churn cycles with dataplane is not a problem in post-processing. (I have a script that passes the timestamps from kube-burner ( with local-index, jobSummary.json contains churnStart and churnEnd timestamps to query the crucible data (or vice-versa)). However, we still keep the intervals between churn cycles in this way of getting the data.

**Comment by vishnuchalla** [comment]
> I am orchestrating PIDs with a python script, e.g. the dataplane starts (it takes +3 min. for the dataplane workload to actually start, with crucible as an example, only after "`client-start-end`"), and then kube-burner is called.

@sferlin looks like you have an alternative already figured out for this. But in case if you see any gaps or something would be nice fit to add into kube-burner, please feel to open up a separate issue.
